modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/RenameContentCommand.java

```diff
@@ -47,63 +55,57 @@ Content execute()
         }
     }
 
-    private Content doExecute()
+    private NodeDataProcessor initProcessors()
     {
-        final NodeId nodeId = NodeId.from( params.getContentId() );
-
-        final NodeName nodeName = NodeName.from( params.getNewName().toString() );
-
-        final RenameNodeParams.Builder builder = RenameNodeParams.create().nodeId( nodeId ).refresh( RefreshMode.ALL ).nodeName( nodeName );
+        final List<NodeDataProcessor> processors = new ArrayList<>();
 
         if ( params.stopInherit() )
         {
-            builder.processor( new RenameContentProcessor() );
+            processors.add( new InheritedContentDataProcessor()
+            {
+                @Override
+                protected EnumSet<ContentInheritType> getTypesToProceed()
+                {
+                    return EnumSet.of( ContentInheritType.NAME );
+                }
+            } );
         }
 
-        final Node node = nodeService.rename( builder.build() );
+        processors.add( ( data, newNodePath ) -> {
+            final Node persistedNode = nodeService.getById( NodeId.from( params.getContentId().toString() ) );
```

Comment:
back to drawing board....
processor must not fetch node as it operates on not yet persisted data.